include/kernel.h

```diff
@@ -2958,6 +2962,53 @@ extern int k_delayed_work_submit_to_queue(struct k_work_q *work_q,
  */
 extern int k_delayed_work_cancel(struct k_delayed_work *work);
 
+/**
+ * @brief Get uptime expiration of a delayed_work
+ *
+ * This routine returns the value of the system uptime counter at
+ * which point a scheduled delayed work item will execute. If the
+ * delayed_work is not scheduled, it returns the current time.
```

Comment:
"Return current time if timeout not set" approach in this API does not sound good for me. This makes timeout expiring now indistinguishable from no timeout set scenario. Could we return K_FOREVER_TICKS or other special value in all getters?

Reply:
Fixed, everything returns K_FOREVER_TICKS as requested.  The "current time" choice was mostly just an optimization, but the few branches needed aren't a big deal.

Reply:
The comment above still states " If the delayed_work is not scheduled, it returns the current time."